pkg/image/decompress.go

```diff
@@ -71,6 +80,14 @@ func gunzip(r io.ReadCloser) (io.ReadCloser, error) {
 	return gzip.NewReader(r)
 }
 
+func xzDecompress(r io.ReadCloser) (io.ReadCloser, error) {
```

Comment:
I like the name; should we change `gunzip` to match the pattern?  Like `gzDecompress`?  That helps denote that they both act similarly I think.